diff --git a/commands/svc.cmd b/commands/svc.cmd
index 1b94d49..e824569 100644
--- a/commands/svc.cmd
+++ b/commands/svc.cmd
@@ -62,7 +62,9 @@ if [[ "${WARDEN_PARAMS[0]}" == "up" ]]; then
 
     ## copy configuration files into location where they'll be mounted into containers from
     mkdir -p "${WARDEN_HOME_DIR}/etc/traefik"
-    cp "${WARDEN_DIR}/config/traefik/traefik.yml" "${WARDEN_HOME_DIR}/etc/traefik/traefik.yml"
+    if (containsElement "-d" "$@" || containsElement "--detach" "$@"); then
+        cp "${WARDEN_DIR}/config/traefik/traefik.yml" "${WARDEN_HOME_DIR}/etc/traefik/traefik.yml"
+    fi
 
     ## generate dynamic traefik ssl termination configuration
     cat > "${WARDEN_HOME_DIR}/etc/traefik/dynamic.yml" <<-EOT
